src/main/java/com/cdancy/bitbucket/rest/domain/repository/Hook.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cdancy.bitbucket.rest.domain.repository;
+
+import com.cdancy.bitbucket.rest.domain.common.Error;
+import com.cdancy.bitbucket.rest.domain.common.ErrorsHolder;
+import com.cdancy.bitbucket.rest.utils.Utils;
+import com.google.auto.value.AutoValue;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import java.util.List;
+
+@AutoValue
+public abstract class Hook implements ErrorsHolder {
+
+    @Nullable
+    public abstract HookDetails details();
+
+    public abstract boolean enabled();
+
+    public abstract boolean configured();
+
+    @SerializedNames({ "details", "enabled", "configured", "errors" })
+    public static Hook create(HookDetails details, String enabled, String configured, @Nullable List<Error> errors) {
+        return new AutoValue_Hook(Utils.nullToEmpty(errors), details, Boolean.valueOf(enabled), Boolean.valueOf(configured));
+    }
```

Comment:
Think you misunderstood ;) Lets do something like this:

```
@SerializedNames({ "details", "enabled", "configured", "errors" })
 public static Hook create(HookDetails details, Boolean enabled, Boolean configured, @Nullable List<Error> errors) {
         return new AutoValue_Hook(Utils.nullToEmpty(errors), details, Boolean.valueOf(enabled), Boolean.valueOf(configured));
 }
```

This way, if those values are null, it will still resolve to false for the actual object creation.